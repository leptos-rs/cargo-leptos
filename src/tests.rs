use camino::Utf8PathBuf;

use crate::{
    config::{BuildOpts, Cli, Commands},
    ext::PathBufExt,
    run,
};

#[tokio::test]
async fn workspace_build() {
    let command = Commands::Build(BuildOpts::default());

    let cli = Cli {
        manifest_path: Some(Utf8PathBuf::from("examples/workspace/Cargo.toml")),
        log: Vec::new(),
        command,
    };

    run(cli).await.unwrap();

    // when running the current working directory is changed to the manifest path.
    let site_dir = Utf8PathBuf::from("target/site");

    //insta::assert_snapshot!(site_dir.ls_ascii(0).unwrap_or_default());
}

// TODO: `cargo-leptos` sets the cwd which is a global env
// and that prevents builds to run in parallel in the same process
//
// #[tokio::test]
// async fn project_build() {
//     let command = Commands::Build(Opts::default());

//     let cli = Cli {
//         manifest_path: Some(Utf8PathBuf::from("examples/project/Cargo.toml")),
//         log: Vec::new(),
//         command,
//     };

//     run(cli).await.unwrap();

//     // when running the current working directory is changed to the manifest path.
//     let site_dir = Utf8PathBuf::from("target/site");

//     insta::assert_snapshot!(site_dir.ls_ascii(0).unwrap_or_default());
// }
