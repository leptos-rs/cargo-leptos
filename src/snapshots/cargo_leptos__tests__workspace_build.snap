---
source: src/tests.rs
expression: site_dir.ls_ascii(0).unwrap_or_default()
---
site:
  project1:
    favicon.ico
    pkg:
      project1.css
      project1.js
      project1.wasm
      snippets:
        leptos_reactive-81881a7e036355c6:
          inline0.js
  project2:
    favicon.ico
    pkg:
      project2.css
      project2.js
      project2.wasm
      snippets:
        leptos_reactive-81881a7e036355c6:
          inline0.js
